oio/cli/admin/service_check.py

```diff
@@ -280,42 +284,35 @@ def formatter_default(self):
     def _format_results(self, checker):
         for res in checker.run():
             if res.target.type == 'chunk':
-                yield repr(res.target)[len('chunk='):], \
-                    res.health,\
-                    res.errors_to_str()
+                if not res.errors:
+                    status = 'OK'
+                else:
+                    status = 'error'
+                    self.success = False
+                yield (repr(res.target)[len('chunk='):],
+                       status, res.errors_to_str())
 
     def get_parser(self, prog_name):
         parser = super(RawxCheck, self).get_parser(prog_name)
-        parser.add_argument(
-            'services',
-            nargs='+',
-            metavar='<service_id>',
-            help=('Rawx service to check.')
-        )
+        self.patch_parser(parser)
         return parser
 
-    def _ensure_http_prefix(self, url):
-        if not url.startswith('http://'):
-            return 'http://' + url
-        return url
+    def take_action(self, parsed_args):
+        self.check_and_load_parsed_args(self.app, parsed_args)
+        return super(RawxCheck, self).take_action(parsed_args)
 
-    def check_services(self, parsed_args):
-        checker = self.build_checker(parsed_args)
+    def _take_action(self, parsed_args):
         for service in parsed_args.services:
             reqid = self.app.request_id(self.reqid_prefix)
             chunks = self.app.client_manager.rdir.chunk_fetch(service,
                                                               reqid=reqid)
-            for res in self.check_chunks(service, chunks, checker):
+            for res in self.check_chunks(service, chunks, self.checker):
                 yield res
 
     def check_chunks(self, service, chunks, checker):
-        url = self._ensure_http_prefix(service)
+        url = 'http://' + service
```

Comment:
Is it compatible with `Service ID` ?

Reply:
Yes, it's compatible with `Service ID`.
https://github.com/open-io/oio-sds/blob/master/oio/blob/cache.py#L53